### DISCUSSION TITLE
FLAC errors

### DISCUSSION THREAD

**Discussion by rnmerchant** [enzo1982/freac#279] [Help & Support] [answered]
Seems I more and more often get processing errors these days ("failed to verify input track: Device..." Checksum error").  I had guessed issues with reading CD so clean-clean-clean-polish and this fixes some but others remain.  However there are encoded files.  Don't seem to get errors when iTunes imports the CD.
Is there anything one should do in addition?  Is it OK having some checksum errors when the files seem to play?.  I see there is a minor update to the FLAC codec - should I try to implement that (and how?)?
Thanks...

**Comment by enzo1982** [comment, marked as answer]
These verification errors come from AccurateRip. It checks the result of a CD rip against an online database to compare it to the results other people got for the same CD tracks.

You don't get verification errors with iTunes, because iTunes does not perform any verification. It just uses what the drive reads without asking any questions...

Also, this doesn't have anything to do with FLAC. What is checked is the data read by the CD drive. This happens before the data is sent to the FLAC encoder.

With AccurateRip you will basically get two kinds of verification errors:

1. Most tracks of a CD are successfully verified, but for some you get a verification error.
This is most likely a case of a read error introduced by your drive. You can often get successful verification for these tracks by cleaning the CD or using another drive.

2. All tracks of a CD fail to be verified.
This is most likely the result of the CD being from a different pressing than the ones in the database. Different CD pressings can have slightly different track offsets which results in different checksums. I plan to add support for detecting this case and turning it into successful verifications in a future fre:ac update.

In the first case, you should try to clean the CD or use a different drive to get a successful verification.

In the second case, you can either ignore the error (which will be the same as ripping with a software that does not support CD rip verification) or make a second rip and compare the results. If the output files of two rips are identical, the rip is probably fine/error free.

You also have the option of disabling AccurateRip verification (under _Options->Configure AccurateRip_). In that case fre:ac will behave just like iTunes or earlier fre:ac versions (before 1.1.5).

**Comment by rnmerchant** [comment]
Thanks.   This all makes sense.  On a couple of occasions I have had several errors - then cleaned and polished the disk - then generally most (or perhaps - but not commonly - all) are gone.  Not a huge deal but nice to understand.

Thanks for the work.

**Comment by DM1785** [comment]
I think I know what is causing this "Failed to verify....." error. I routinely get it on an entire CD. I just figured out that it is happening only when I add to or change tag data or track information. Even if I submit the data, I get the error. If I rip a CD without editing tags, track titles, artist or whatever, I get no error(s). The most irritating thing is that fre:ac does not create play lists or cue sheets when this error occurs. The flac files all play OK when I do get the errors.
